src/raw.rs

```diff
@@ -232,18 +232,59 @@ pub fn string_dma(key: *mut RedisModuleKey, len: *mut size_t, mode: KeyMode) ->
     unsafe { RedisModule_StringDMA.unwrap()(key, len, mode.bits) }
 }
 
-pub fn hash_get(key: *mut RedisModuleKey, field: &str) -> *mut RedisModuleString {
-    let res: *mut RedisModuleString = ptr::null_mut();
-    unsafe {
-        RedisModule_HashGet.unwrap()(
-            key,
-            REDISMODULE_HASH_CFIELDS as i32,
-            CString::new(field).unwrap().as_ptr(),
-            &res,
-            ptr::null::<c_char>(),
-        );
+pub fn hash_get_multi<T>(key: *mut RedisModuleKey, fields: &[T],
+                         values: &mut [*mut RedisModuleString])
+    -> Result<(), RedisError>
+    where T: Into<Vec<u8>> + Clone
```

Comment:
@slavak what does the generic parameter `T` provide? Why not simply use `fields: &[&str]` instead of `&[T]`?
If you see a need to provide e.g. `&[&String]`, you can use `AsRef<str>` as a bound.

Using `&[&str]` also allows getting ride of the `clone()` call in `CString::new` below (provided you use `let mut fi = fields.into_iter()`).

Reply:
`fields: &[&str]` would prevent you from passing in a `&[String]` without allocating a new vector, which is unnecessarily limiting. `Into<Vec<u8>> + Clone` is just the minimal bound we can specify here. (Required because that's the bound for `CString::new`.)

The `clone` is necessary because iterating over a slice returns a borrowed reference for both `iter` and `into_iter`, while `CString::new` takes ownership of its argument. For slices of `&str` this should be very cheap because `&str` is `Copy`. For slices of `String` it's unfortunately a necessary cost unless we want to take ownership of the data, but I'm not sure how to do that in a way that's ergonomic.

Reply:
Incidentally for `&[&str]` the clone shouldn't do any additional work because it's equivalent to the copy that already happens implicitly.

Reply:
Sounds good to me.